src/Plugin.php

```diff
@@ -113,6 +126,12 @@ public function logDebug($message)
 
     public function handleIrcReceived(UserEvent $event, EventQueue $queue)
     {
+        if (
+            $this->filter !== null &&
+            $this->filter->filter($event) !== false
+        ) {
+            return;
+        }
```

Comment:
You're right, currently writing a set of filters that lets user filter based on hostname, path, protocol, and port.